spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java

```diff
@@ -116,4 +117,16 @@ private void registerMessagingAnnotationPostProcessors(BeanDefinitionRegistry re
 		}
 	}
 
+	private void registerGatewayProxyInstantiationPostProcessor(BeanDefinitionRegistry registry) {
+		if (!registry.containsBeanDefinition("gatewayProxyBeanDefinitionPostProcessor")) {
+			BeanDefinitionBuilder builder =
+					BeanDefinitionBuilder.genericBeanDefinition(GatewayProxyInstantiationPostProcessor.class,
+							() -> new GatewayProxyInstantiationPostProcessor(registry))
+							.addConstructorArgValue(registry)
```

Comment:
You don't need both of these (instance supplier + constructor arg).
```suggestion
							.addConstructorArgValue(registry)
```

Reply:
I think we do need to AOT generation.
The supplier works in for regular JVM, but AOT will generate for us an new bean definition based on the args provided: there is no way in a generated code to get access to those suppliers.
Do I miss anything?
If you want I can verify that a bit later.
Thanks

Reply:
I understand that, but why bother using a supplier for the regular JVM when just the class and ctor arg is needed?